tests/PyPI-pytest/run_all.py

```diff
@@ -0,0 +1,178 @@
+                args=[
+                    os.path.join(
+                        venv.getVirtualenvDir(),
+                        "bin" if os.name != "nt" else "scripts",
+                        "python",
+                    ),
+                    "-m",
+                    "pytest",
+                    "--disable-warnings",
+                ],
+                stdout=subprocess.PIPE,
+                stderr=subprocess.PIPE, 
+            )
+            
+            stdout, stderr = process.communicate()
+            exit_pytest = process.returncode
+            print("stdout: \n", stdout)
+            print("stderr: \n", stderr)
+            
+    except Exception as exceptObj:
+        print(exceptObj)
+        continue
+
+
+    os.chdir(base_dir)
+
+    try:
+        # get wheel pytest results
+        with withVirtualenv("wheel_venv_%s" % package_name) as venv:
+            venv.runCommand(
+                commands=[
+                    "call Scripts/activate",
+                    "python -m pip install pytest",
+                    "git clone https://github.com/%s/%s.git" % (package_name,package_name),
+                    "git clone https://github.com/nuitka/nuitka.git",
+                    "cd nuitka",
+                    "python setup.py develop",
+                    "cd ../%s" % package_name,
+                    "python -m pip install -r dev-requirements.txt",
+                    "python setup.py bdist_nuitka",
+                    "python -m pip uninstall -y %s" % package_name,
+                ]
+            )
+
+            dist_dir = os.path.join(base_dir, "wheel_venv_%s" % package_name, package_name, "dist")
+
+            venv.runCommand(
+                'pip install "%s"'
+                % (os.path.join(dist_dir, os.listdir(dist_dir)[0]))
+            )
+
+            os.chdir(os.path.join(base_dir, "wheel_venv_%s" % package_name, package_name))
+
+            process = subprocess.Popen(
+                args=[
+                    os.path.join(
+                        venv.getVirtualenvDir(),
+                        "bin" if os.name != "nt" else "scripts",
+                        "python",
+                    ),
+                    "-m",
+                    "pytest",
+                    "--disable-warnings",
+                ],
+                stdout=subprocess.PIPE,
+                stderr=subprocess.PIPE, 
+            )
+
+            w_stdout, w_stderr = process.communicate()
+            w_exit_pytest = process.returncode
+            print("stdout: \n", stdout)
+            print("stderr: \n", stderr)
+    
+    except Exception as exceptObj:
+        print(exceptObj)
+        continue
+
+
+    os.chdir(base_dir)
+    # remove venv directory
+    # currently not removing the directories?
+    # removeDirectory(os.path.join(base_dir, "venv_%s" % package_name), ignore_errors=False)
+    # removeDirectory(os.path.join(base_dir, "wheel_venv_%s" % package_name), ignore_errors=False)
```

Comment:
`removeDirectory` is failing to remove the directories, giving a `PermissionError`, my guess is it has to do with the virtualenvs?

Reply:
The `PermissionError` happens a lot on Windows, after you ran a binary.

You cannot delete a binary that has been executed, because you do not know when it really exits, and what happens a lot too, is that Virus checkers work too.

I have excluded my Buildbot working stage from `Windows Defender` and that makes it even faster. The issue is transient, is it?

Reply:
I see, is there a way around it? Right now I am manually deleting the virtualenv directories after each run but this deleting should be automated.